contribution/jackson-coding-style-guide.md

```diff
@@ -0,0 +1,69 @@
+# Jackson Coding Style Guide
+
+This document serves as the coding conventions for Java source code in the Jackson project.
+Some parts of this document only apply to Java source code, while others apply to all source code in the project.
+
+## Table of Contents
+
+1. [Source Formatting](#source-formatting)
+    1. [Indentation](#indentation)
+    2. [Import Statements](#import-statements)
+        1. [Wildcard Imports](#wildcard-imports)
+        2. [Ordering](#ordering)
+2. [Naming Conventions](#naming-conventions)
+    1. [Field Naming](#field-naming)
+    2. [Method Naming](#method-naming)
+
+## Source Formatting
+
+### Indentation
+
+- **Rule**: Use spaces instead of tabs.
+- **Amount**: 4 spaces per indentation level.
+
+### Import Statements
+
+#### Wildcard Imports
+
+- **Usage**: Wildcards are permissible.
+- **Threshold**: Employ wildcards for 5 or more imports from the same package.
+- **Exceptions**: For frequently used packages like `java.util`, `java.io`, and main Jackson packages (
+  e.g., `com.fasterxml.jackson.databind`), wildcards may be used for fewer imports.
+
+#### Ordering
+
+Import statements should be grouped and ordered in the following manner:
+
```

Comment:
In addition, I think I have some thoughts on static imports. F.ex:

1. Static imports should be used sparingly: their use can reduce readability of code
2. When used, static imports should come before non-static imports, but using same ordering (from most generic (JDK) to most specific (component itself) ) as static imports.

(NOTE: I used to prefer static imports coming after non-static one -- I can be convinced either way, and the main thing is to agree one way to go and stick to it)

Reply:
Noted, will make some changes! Thank you for the feedback. 🙏🏼

Reply:
If static imports should come "before" regular imports, it would be because we should be using less static things so increase visibility.

Reply:
+1 on static coming first. It would be nice to hear what others think.

Reply:
i think static imports should come after because that's the intellij default :)

Reply:
Right, IntelliJ automatic imports 🙉 I am pretty sure IntelliJ won't throw import statements to where we intend, but at least near by :)))

Reply:
As I said, I do not have a strong opinion. But I think Eclipse also adds them after other import statements, fwtw.

Then again, IDEs default to (I think) alphabetic ordering of imports which is also not how Jackson expects them to be sorted.

Reply:
i think we all thinking same now.

- `non-static` then `static`
- ordering follows Jackson style (java -> 3rd party -> jackson)